src/pages/VerificationPage/useVerification.ts

```diff
@@ -0,0 +1,111 @@
+import { Dispatch, SetStateAction, useEffect, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { useLocation, useNavigate } from 'react-router-dom';
+import { toast } from 'react-toastify';
+
+import { sendVerificationCode, verifyCode } from '@/api/userActions';
+import { useAppDispatch } from '@/redux/hooks';
+
+interface UseVerificationReturnType {
+  otp: string;
+  setOtp: Dispatch<SetStateAction<string>>;
+  timeLeft: number;
+  focusedIndex: number | null;
+  setFocusedIndex: Dispatch<SetStateAction<number | null>>;
+  handleResendCode: () => Promise<void>;
+  handleSubmit: () => Promise<void>;
+  email: string;
+  isLocked: boolean;
+}
+
+const useVerification = (
+  otpExpiryTime: number,
+  numInputs: number
+): UseVerificationReturnType => {
+  const navigate = useNavigate();
+  const location = useLocation();
+  const [otp, setOtp] = useState<string>('');
+  const [timeLeft, setTimeLeft] = useState<number>(otpExpiryTime);
+  const [focusedIndex, setFocusedIndex] = useState<number | null>(null);
+  const [attemptCount, setAttemptCount] = useState(0);
+  const [isLocked, setIsLocked] = useState(false);
+  const dispatch = useAppDispatch();
+  const { t } = useTranslation();
+
+  const email: string = location.state?.email || '';
+
+  const tick = 1000;
+  const maxTries = 5;
```

Comment:
move above useEffect

Reply:
I can't move above useEffect because variables that i use inside the useEffect must be above